app/models/course.rb

```diff
@@ -15,7 +15,7 @@ class Course < ApplicationRecord
   # rubocop:todo Rails/HasManyOrHasOneDependent
   has_many :media, -> { order(position: :asc) },
            as: :teachable,
-           inverse_of: course
+           inverse_of: :course
```

Comment:
Note that this is (while syntactically correct now) still not correct. Calling course.media now gives an exception:

```
/usr/local/bundle/gems/activerecord-7.0.4.3/lib/active_record/reflection.rb:236:in `check_validity_of_inverse!': Could not find the inverse association for media (:course in Medium) (ActiveRecord::InverseOfAssociationNotFoundError)
```

Reply:
I changed it in the latest commit to `inverse_of: :teachable`.
This seems to work. See also [here](https://makandracards.com/makandra/47200-association-to-polymorphic-model-does-not-determine-inverse_of-automatically), although in contrast to what is written there, saving also works without the additional` inverse_of`.

Reply:
Oh ok, finally a use case where one writes something different than just the class name. In my head, I've updated the rule to the following mantra, is it correct?

```sh
If "as:" association is specified:
Write inverse_of: :<class that "as:" refers to>
return

If foreign_key is specified, e.g. foreign_key: classname_id
Write inverse_of: :<classname (without the _id part)>
return

If file-level class name ends with Join
Write inverse_of: <convert classname to small snake-case and append an "s" to "join" at the end>
return

Otherwise:
Write inverse_of: :<convert classname to small snake-case>
```

As far as I know, these rules are in accordance with what you've done in c3abe9f82 and 2a730ffe. I've also checked that all `inverse_of` occurrences in the codebase (21 in total) are in line with this ruleset.

I begin to understand why Ruby cannot do this automatically. There might be a lot more edge cases ;)

Reply:
The mantra seems correct to me.